sklearn/ensemble/stacking_classifier.py

```diff
@@ -0,0 +1,319 @@
+    >>> from sklearn.ensemble import RandomForestClassifier, StackingClassifier
+    >>> clf1 = LogisticRegression(random_state=1)
+    >>> clf2 = RandomForestClassifier(random_state=1)
+    >>> clfm = GaussianNB()
+    >>> X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]])
+    >>> y = np.array([1, 1, 1, 2, 2, 2])
+    >>> eclf1 = StackingClassifier(estimators=[
+    ...         ('lr', clf1), ('rf', clf2)], meta_estimator=clfm, cv=2)
+    >>> eclf1 = eclf1.fit(X, y)
+    >>> print(eclf1.predict(X))
+    [1 1 1 2 2 2]
+    >>>
+    """
+
+    def __init__(self, estimators, meta_estimator, method='auto',
+                 cv=None, n_jobs=1, verbose=0):
+        self.estimators = estimators
+        self.named_estimators = dict(estimators)
+        self.meta_estimator = meta_estimator
+        self.cv = cv
+        self.method = method
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+
+    def _method(self, estimator):
+        if self.method == 'auto':
+            if hasattr(estimator, 'predict_proba'):
+                method = 'predict_proba'
+            elif hasattr(estimator, 'decision_function'):
+                method = 'decision_function'
+            else:
+                raise AttributeError("Estimator %s has no method "
+                                     "`predict_proba` or `decision_function`. "
+                                     "Try specify a method instead of using "
+                                     "the default `auto`" % estimator)
+        else:
+            method = self.method
+        return method
+
+    def fit(self, X, y, **kwargs):
+        """ Fit the estimators.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        **kwargs: optional fit parameters
+
+
+        Returns
+        -------
+        self : object
+        """
+        if any(s in type_of_target(y) for s in ['multilabel', 'multioutput']):
+            raise NotImplementedError('Multilabel and multi-output'
+                                      ' classification is not supported.')
+
+        if not self.estimators:
+            raise AttributeError('Invalid `estimators` attribute %s, '
+                                 '`estimators` should be a list of (string, '
+                                 'estimator) tuples.' % type(self.estimators))
+
+        if not is_classifier(self.meta_estimator):
+            raise AttributeError('Invalid `meta_estimator` attribute '
+                                 '%s, `meta_estimator` should be a classifier.'
+                                 % type(self.meta_estimator))
+        for name, step in self.estimators:
+            if not hasattr(step, self._method(step)):
```

Comment:
Unfortunately, at least from a theoretical standpoint, you can't determine this until after fit: a grid search may find an estimator that does or does not support `predict_proba`. Pretty nasty, I'll admit.